### PR TITLE
docs: Rename "unsecure" to insecure

### LINKED ISSUE #269: [BUG]: "Unsecure" is really insecure
### Pre-submission checks

- [X] I am **not** filing a feature request. These should be filed via the feature request form instead.
- [X] I have looked through the [open issues](https://github.com/woodruffw/zizmor/issues?q=is%3Aissue+is%3Aopen+) for a duplicate report.

### Expected behavior

https://github.com/search?q=repo%3Awoodruffw%2Fzizmor%20unsecure&type=code

### Actual behavior

https://github.com/search?q=repo%3Awoodruffw%2Fzizmor+insecure&type=code

### Reproduction steps

Would you merge a renaming PR?

### PR DESCRIPTION
**Pull request by szepeviktor**
Closes  #269

### CODE REVIEW

File: docs/audits.md [hunk 1]
```diff
@@ -634,7 +634,7 @@ Workflow commands (like `::set-env` and `::add-path`)
 to inject environment variables and therefore obtain code execution).
 
 However, users can explicitly re-enable them by setting the
-`ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable at the workflow,
+`ACTIONS_ALLOW_INSECURE_COMMANDS` environment variable at the workflow,
```
Comment (woodruffw) [inline docs/audits.md:637]: We can't change the environment variable anywhere, only the other uses of "unsecure" -- the envvar is GitHub's own misspelling, which we need to detect.

(Sorry, I didn't notice this in your search earlier.)

Comment (szepeviktor) [reply docs/audits.md]: Okay!

Done 🍏

We have 1×unsecure, I've added some "GitHub"-s.

Comment (woodruffw) [reply docs/audits.md]: Thanks!

**Review by woodruffw — APPROVED** [review]
LGTM, thank you @szepeviktor!

**Comment by szepeviktor** [comment]
Glad to contribute.

**Pull request merged**